internal/provider/compliance_framework_resource.go

```diff
@@ -0,0 +1,186 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	mondoov1 "go.mondoo.com/mondoo-go"
+)
+
+var _ resource.Resource = (*complianceFrameworkResource)(nil)
+
+func NewComplianceFrameworkResource() resource.Resource {
+	return &complianceFrameworkResource{}
+}
+
+type complianceFrameworkResource struct {
+	client *ExtendedGqlClient
+}
+
+type complianceFrameworkResourceModel struct {
+	// scope
+	SpaceId types.String `tfsdk:"space_id"`
+
+	// resource details
+	FrameworkMrn types.List `tfsdk:"framework_mrn"`
+	Enabled      types.Bool `tfsdk:"enabled"`
+}
+
+func (r *complianceFrameworkResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_framework_assignment"
+}
+
+func (r *complianceFrameworkResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: `Set Compliance Frameworks for a Mondoo Space.`,
+		Attributes: map[string]schema.Attribute{
+			"space_id": schema.StringAttribute{
+				MarkdownDescription: "Mondoo Space Identifier.",
+				Required:            true,
+			},
+			"framework_mrn": schema.ListAttribute{
+				MarkdownDescription: "Compliance Framework MRN.",
+				Required:            true,
+				ElementType:         types.StringType,
+			},
+			"enabled": schema.BoolAttribute{
+				MarkdownDescription: "Enable or disable the Compliance Framework.",
+				Required:            true,
+			},
+		},
+	}
+}
+
+func (r *complianceFrameworkResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*mondoov1.Client)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *http.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = &ExtendedGqlClient{client}
+}
+
+func (r *complianceFrameworkResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data complianceFrameworkResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	scopeMrn := ""
+	if data.SpaceId.ValueString() != "" {
+		scopeMrn = spacePrefix + data.SpaceId.ValueString()
+	}
+
+	var frameworkList []mondoov1.String
```

Comment:
I recommend to either move this logic into gql with a call like "BulkUpdateFramework" or create a function in this struct that updates all frameworks. Right now we have to copy the code between create and update. Its easy to forget one thing during bug fixing.